pymc_marketing/clv/models/beta_geo.py

```diff
@@ -344,3 +344,61 @@ def expected_num_purchases_new_customer(
         return (left_term * right_term).transpose(
             "chain", "draw", "t", missing_dims="ignore"
         )
+
+    def probability_of_n_purchases_up_to_time(
+        self,
+        t: Union[np.ndarray, pd.Series],
+        n: int,
+    ):
+        r"""
+        Compute the probability of n purchases.
+         .. math::  P( N(t) = n | \text{model} )
+        where N(t) is the number of repeat purchases a customer makes in t
+        units of time.
+        Comes from equation (8) of [2]_.
+        Parameters
+        ----------
+        t: float
+            number units of time
+        n: int
+            number of purchases
```

Comment:
We can think of vectorizing this function for vector-valued `t` and `n`, but probably not in this PR. An issue to keep track of this idea for later 🙂